lib/puppet/util/windows/file.rb

```diff
@@ -89,10 +89,31 @@ def symlink(target, symlink)
   end
   module_function :symlink
 
+
+  def file_exist(file_name)
+    path = file_name.to_s
+    attempts = 10
```

Comment:
You could also maintain a list of seen links and if you revisit a seen one, `raise`.

If you want to use a bounded loop, the canonical Ruby way is with `0.upto(10)`